plugins/update_last_edited.py

```diff
@@ -0,0 +1,47 @@
+import sublime
+import sublime_plugin
+import time
+
+# Related Reading:
+#     https://forum.sublimetext.com/t/automatically-updated-timestamp/7156
+#
+# This is a simple plugin for people that like to have a last edited timestamp
+# directly inside their files. It implements a command that will find such a
+# timestamp based on a header and then update it with the current time. The
+# header to look for and the date format are user customizable.
+#
+# The event listener will apply the command on every save, making this seamless
+# as an operation. If you would rather do this manually, remove or comment out
+# the event handler class and you can bind the command to a key instead.
+#
+# This example binding uses the chain command built into Sublime Text 4 to
+# execute first this command, then the save command.
+
+# { "keys": ["super+t"], "command": "chain",
+#    "args": {
+#       "commands": [
+#          {"command": "update_last_edited_date"},
+#          {"command": "save"},
+#       ]
+#    }
+# },
+
+class UpdateLastEditedDateCommand(sublime_plugin.TextCommand):
+    """
+    Check the current file for the first line that contains the provided
+    header, and if found update it to include the current date and time
+    as formatted by the given date format.
+    """
+    def run(self, edit, hdr="Last Edited: ", fmt="%d %b %Y %I:%M%p"):
+        span = self.view.find(f'{hdr}.*$', 0)
```

Comment:
it may make sense to regex escape the `hdr`. Also, curious why not call it `header`? :)

Reply:
The short name was entirely for brevity reasons so that it would fit in a code block in the forum without scrolling. I'm cool with changing it to a longer name generally though.

Reply:
Not sure I get why this would benefit from a regex escape; doesn't that only matter for constant strings and not ones generated at runtime? I have expanded out the names of the arguments to be longer and more readable though.

Reply:
I'm thinking that if the command is run with a user-supplied `header` argument as opposed to using the default value defined here, it could consist of anything - parens, or other characters which have special meaning in regex, and if we want to treat it as a literal, it would need to be escaped? To be clear, its not about escaping for the f-string, but the regex engine - otherwise after the f-string is resolved, if the header was `*** last modification: `, the regex pattern could be e.g. `*** last modification: .*$` and would have different meaning than if we escaped the asterisks: `\*\*\* last modification: .*$`

Am I making sense? :)

Reply:
Total sense! I've added in some quoting action on the header string as it comes in. It seems to be a bit overzealous, such as quoting spaces which seems unnecessary, but since I didn't notice this as an issue clearly I'm a boob anyway.